index.js

```diff
@@ -0,0 +1,206 @@
+        headers: {
+          'Content-Type': 'application/json',
+          'User-Agent': 'metal-github-runner-action',
+          'X-Auth-Token': authToken
+        },
+        method: 'GET'
+      };
+
+      const req = https.request(options, res => {
+        let response = '';
+
+        res.on('data', chunk => {
+          response += chunk;
+        });
+
+        res.on('end', error => {
+          if (res.statusCode >= 200 && res.statusCode < 300) {
+            const state = JSON.parse(response).state;
+            if (
+              state == 'failed' ||
+              state == 'deprovisioning' ||
+              state == 'deleted'
+            ) {
+              error = new Error(
+                'Server provisioning failed' + `State: ${state}`
+              );
+              reject(error);
+            }
+            resolve(state);
+          } else {
+            error = new Error('Request Failed.\n' + `Response: ${response}`);
+            reject(error);
+          }
+        });
+      });
+
+      // Handle Error
+      req.on('error', error => {
+        error = new Error(`Got error: ${error.message}`);
+        reject(error);
+      });
+      req.end();
+    });
+
+    return serverStatus;
+  } catch (error) {
+    core.setFailed(`Server provisioning failed: ${error.message}`);
+  }
+}
+
+async function getIPAddress(serverId) {
+  try {
+    const ipAddress = await new Promise((resolve, reject) => {
+      const options = {
+        hostname: 'api.equinix.com',
+        path: `/metal/v1/devices/${serverId}`,
+        headers: {
+          'Content-Type': 'application/json',
+          'User-Agent': 'metal-github-runner-action',
+          'X-Auth-Token': authToken
+        },
+        method: 'GET'
+      };
+
+      const req = https.request(options, res => {
+        let response = '';
+
+        res.on('data', chunk => {
+          response += chunk;
+        });
+
+        res.on('end', error => {
+          if (res.statusCode >= 200 && res.statusCode < 300) {
+            const ipAddress = JSON.parse(response).ip_addresses[0].address;
+            resolve(ipAddress);
+          } else {
+            error = new Error('Request Failed.\n' + `Response: ${response}`);
+            reject(error);
+          }
+        });
+      });
+
+      // Handle Error
+      req.on('error', error => {
+        error = new Error(`Got error: ${error.message}`);
+        reject(error);
+      });
+      req.end();
+    });
+
+    return ipAddress;
+  } catch (error) {
+    core.setFailed(`Failed to get IP Address: ${error.message}`);
+  }
+}
+
+function sleep(ms) {
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+async function run() {
+  try {
+    const serverId = await createServer();
+    let serverStatus = '';
+    while (serverStatus !== 'active') {
+      serverStatus = await checkStatus(serverId);
+      core.info(`Server Status: ${serverStatus}...`);
+      await sleep(5000);
```

Comment:
Not a big deal for now, but we might want a higher-level timeout (unless there already is one and I'm missing it) so that the action fails if the server still hasn't become active after, e.g., 30 minutes.

Reply:
Was hoping to get a suggestion on how best to do that.

Reply:
It'd be something like:
```javascript
// before the while loop
timeoutInMillis = 1.8e6;
timeoutID = setTimeout(function() {
  throw new Error('device did not become active within 30 minutes');
}, timeoutInMillis);

// after the while loop
clearTimeout(timeoutID);
```
Reference docs for setTimeout are here: https://developer.mozilla.org/en-US/docs/Web/API/setTimeout

Reply:
Another option is [`Promise.race()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/race).